### PR TITLE
Rdiscrowd 4900 cache warmup issue showing in the RQ

### PR DESCRIPTION
**Pull request by XiChenn**
*Issue number of the reported bug or feature request: #Rdiscrowd 4900*

**Describe your changes**
Add request_context() in the functions of jobs.py so that it has "request" object available. This is to avoid the error "RuntimeError: Working outside of request context."

**Testing performed**
Tested in QA

### CODE REVIEW

**Comment by XiChenn** [comment]
> Thank you for the PR. In addition to resolving the error of "operating outside of request context", does this also address any performance issues? Thanks.

This can improve the READ performance, but it could have some negative impacts on the WRITE performance, as those two supposed running functions have lots of "UPDATE project_stats" table actions. Although many write, it's updating different records. So I think the pros should be greater than the cons. We will need to test/observer it in the QA.

File: pybossa/jobs.py [hunk 1]
```diff
@@ -385,10 +387,11 @@ def get_autoimport_jobs(queue='low'):
 @with_cache_disabled
 def get_project_stats(_id, short_name):  # pragma: no cover
     """Get stats for project."""
-    import pybossa.cache.project_stats as stats
+    with current_app.request_context(DUMMY_ENVIRON):
```
Comment (ywrsusan) [inline pybossa/jobs.py:390]: Why do we want to request dummy context instead of the real context?

Comment (XiChenn) [reply pybossa/jobs.py:390]: > 

It's a job so it's not providing an API and there is not real HTTP connection. But, update_stats-> stats_format_dates-> gettext->...(some flask lib) -> _get_locale, and we use "request" object in the [_get_locale](https://github.com/bloomberg/pybossa/blob/0b52645f618661ead7ff0d869218bd9a9cfe9e25/pybossa/core.py#L340). It is a workaround.

Comment (ywrsusan) [reply pybossa/jobs.py:390]: Is it more standard to create a context instead? 
```
app = create_app(run_as_server=False)
```
and do similar thing
```
with app.app_context():
       # do something.
```

Comment (XiChenn) [reply pybossa/jobs.py:390]: > 

If you check the commit history of this pr, you will find my first [commit](https://github.com/bloomberg/pybossa/pull/678/commits/1dfc0e94b963b013eeb6516accb69e7a47442aec) was trying to change in this way. It was proved not working in QA.

Comment (ywrsusan) [reply pybossa/jobs.py:390]: Do you know why it wouldn't work? it seems to be working in other code, e.g. https://github.com/bloomberg/pybossa/blob/e3114b3f719575aec9920cacfba3ebc9e619ca5e/cli.py#L151

Comment (XiChenn) [reply pybossa/jobs.py:390]: If this works it is probably because none of the function finally calls any function using "request" object.

**Review by ywrsusan — APPROVED** [review]
thanks

**Pull request merged**